engines/bops_core/app/controllers/concerns/bops_core/magic_link_authenticatable.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module BopsCore
+  module MagicLinkAuthenticatable
+    extend ActiveSupport::Concern
+
+    included do
+      rescue_from ActionController::ParameterMissing do |exception|
+        render plain: "Not Found", status: :not_found
+      end
+    end
+
+    def authenticate_with_sgid!
+      resource = sgid_authentication_service.locate_resource
+
+      handle_expired_or_invalid_sgid if resource.nil?
+    end
+
+    private
+
+    def sgid_authentication_service
+      @sgid_authentication_service ||= SgidAuthenticationService.new(sgid)
+    end
+
+    def sgid
+      params.require(:sgid)
+    end
+
+    def handle_expired_or_invalid_sgid
+      expired_resource = sgid_authentication_service.expired_resource
+
+      if expired_resource
```

Comment:
since this doesn't seem to get used again does it need to be assigned to a variable?

```suggestion
      if sgid_authentication_service.expired_resource
```